src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests2.cs

```diff
@@ -632,5 +635,1498 @@ public static class E
             // lock (x) { }
             Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "x").WithArguments("System.Threading.Lock+Scope", "Dispose").WithLocation(2, 7));
     }
+
+    [Fact]
+    public void Nullability_PropertyAccess_01()
+    {
+        // nullability check on the receiver, annotated extension parameter
+        var src = """
+#nullable enable
+
+object? oNull = null;
+_ = oNull.P;
+
+object? oNull2 = null;
+E.get_P(oNull2);
+
+object? oNotNull = new object();
+_ = oNotNull.P;
+
+E.get_P(oNotNull);
+""";
+        var libSrc = """
+#nullable enable
+
+public static class E
+{
+    extension(object? o)
+    {
+        public int P { get => throw null!; }
+    }
+}
+""";
+        var comp = CreateCompilation([src, libSrc]);
+        comp.VerifyEmitDiagnostics();
+
+        var libComp = CreateCompilation(libSrc);
+        var comp2 = CreateCompilation(src, references: [libComp.EmitToImageReference()]);
+        comp2.VerifyEmitDiagnostics();
+    }
+
+    [Fact]
+    public void SynthesizedAttributeOnParameters_Params_01()
```

Comment:
>SynthesizedAttributeOnParameters_Params_01 [](http://example.com/codeflow?start=16&length=42)

I think that for synthesized attributes we should test them directly. I.e. asserting their presence in metadata instead of only inferring their presence indirectly from behavior.